### ISSUE TITLE
Error setting up player: Invalid license key

### ISSUE THREAD

**Issue by AppCraft-LLC** [jwplayer/jwplayer-sdk-ios-demo#20]
Found the following issue: JWPlayerController occasionally returns "Error setting up player: Invalid license key" when being used in UICollectionView. For example 40 of 50 players in the flow (UICollectionView) loads and plays streams correctly, but other 10 returns this error and doesn't load content. In this example 3 player of 30 returned that error and didn't load videos: 

`
2017-03-15 09:37:07.496 ST[78893:4898107] SETUP ERROR {
    event = onSetupError;
    message = "Error setting up player: Invalid license key";
}
2017-03-15 09:37:07.525 ST[78893:4898107] SETUP ERROR {
    event = onSetupError;
    message = "Error setting up player: Invalid license key";
}
2017-03-15 09:37:07.529 ST[78893:4898107] SETUP ERROR {
    event = onSetupError;
    message = "Error setting up player: Invalid license key";
}
`

All players (cells of uicollectionview) use the same code to load and play video, but it works in 80-90% and fails on other cases.  We suppose that the error began occurring after we start using - (void)loadConfig:(JWConfig *)config; method rather than initialisation all players from scratch for optimisation purposes. Not sure about this, but if we load 10-30 players simultaneously in a UICollectionView without reusing players pool - everything works fine. Much more slower and uses much more RAM, but works without errors.

**Comment by karimJWP** [comment]
How are you setting your key ? In the latest version we introduced setting the key with a class method called 'setPlayerKey:', which you can use once in your app delegate's 'didFinishLaunchingWithOptions:'. We will look into a fix, in the meantime could you try that ?

**Comment by karimJWP** [comment]
also, if all you are changing is the file I recommend using 'load:' or 'loadPlaylist:'

**Comment by AppCraft-LLC** [comment]
Thank you for your response, yes, the license key was set up correctly. I think that the issue was in the architecture of the app I working on: I used IGListKit to show a flow of the videos and made some cache mechanics to load streams. After I turned off caching (instead of reusing players I always construct new ones when cells entering working area) the error about wrong license key gone and never appears. So I suppose that using loadConfig: with pool of 20-30 players occasionally produces that error. 

But now with new mechanics another error appears: "Cannot Decode". It appear much more rarely that the previous error (approx 1-2 errors for 50 loadings), but when the error appears it influence on images too (for example if I cache posters requesting https://content.jwplatform.com/thumbs/{key}-720.jpg for other videos in the same time when the error "JPEGDecompressSurface : Picture decode failed: e00002d1" appears). It seems like the app overuse the API or something like that.
